src/test/java/no/sikt/nva/pubchannels/handler/CreateJournalHandlerTest.java

```diff
@@ -210,6 +215,108 @@ void shouldReturnBadGatewayWhenAuthClientInterruptionOccurs() throws IOException
         assertThat(problem.getDetail(), is(equalTo("Unable to reach upstream!")));
     }
 
+    @ParameterizedTest
+    @MethodSource("invalidNames")
+    void shouldReturnBadRequestWhenNameInvalid(String name) throws IOException {
+
+        var testJournal = new CreateJournalRequestBuilder().name(name).build();
+        handlerUnderTest.handleRequest(constructRequest(testJournal), output, context);
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
+
+        var problem = response.getBodyObject(Problem.class);
+        assertThat(problem.getDetail(), is(containsString("String is too")));
+    }
+
+    @ParameterizedTest
+    @MethodSource("invalidIssn")
+    void shouldReturnBadRequestWhenInvalidPissn(String issn) throws IOException {
+        var testJournal = new CreateJournalRequestBuilder().name(VALID_NAME)
+                .pissn(issn).build();
+        handlerUnderTest.handleRequest(constructRequest(testJournal), output, context);
+        var response = GatewayResponse.fromOutputStream(output, Problem.class);
+
+        assertThat(response.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_BAD_REQUEST)));
+
+        var problem = response.getBodyObject(Problem.class);
+        assertThat(problem.getDetail(), is(containsString("Issn has an invalid format")));
```

Comment:
I would prefer to inform which issn field is not valid. In this example: "Print issn/pissn has an invalid format"

Reply:
I second that.